### PR TITLE
GEOMESA-1599 Package SparkSQL in dist

### PR DESCRIPTION
**Pull request by anthonyccri**
* Added Spark SQL components to the distribution
* Shaded thrift 0.9.3 for Accumulo 1.8 support (Spark uses conflicting 0.9.2)

Signed-off-by: Anthony Fox <anthonyfox@ccri.com>

### CODE REVIEW

File: geomesa-accumulo/geomesa-accumulo-dist/pom.xml [hunk 1]
```diff
@@ -26,9 +26,18 @@
         <!-- Spark (dist)-->
         <dependency>
             <groupId>org.locationtech.geomesa</groupId>
-            <artifactId>geomesa-accumulo-compute_${scala.binary.version}</artifactId>
+            <artifactId>geomesa-spark-accumulo_${scala.binary.version}</artifactId>
+            <version>${project.version}</version>
```
Comment (elahrvivaz) [inline geomesa-accumulo/geomesa-accumulo-dist/pom.xml]: should we put this in the main pom along with the others?

File: geomesa-accumulo/geomesa-accumulo-dist/src/main/assemblies/component.xml [hunk 2]
```diff
@@ -60,7 +60,7 @@
             <useProjectArtifact>false</useProjectArtifact>
             <useTransitiveDependencies>false</useTransitiveDependencies>
             <includes>
-                <include>org.locationtech.geomesa:geomesa-accumulo-compute*</include>
+                <include>org.locationtech.geomesa:geomesa-spark*</include>
```
Comment (elahrvivaz) [inline geomesa-accumulo/geomesa-accumulo-dist/src/main/assemblies/component.xml]: we probably don't want to bundle the generic data store one here - not sure about all others

File: geomesa-spark/geomesa-spark-accumulo/pom.xml [hunk 3]
```diff
@@ -101,6 +105,11 @@
                                     <pattern>org.apache.commons.codec</pattern>
                                     <shadedPattern>org.locationtech.geomesa.shaded.org.apache.commons.codec</shadedPattern>
                                 </relocation>
+                                <!-- shade thrift because spark depends on 0.9.2 and accumulo 1.8 depends on 0.9.3 -->
+                                <relocation>
+                                    <pattern>org.apache.thrift</pattern>
+                                    <shadedPattern>org.lt.geomesa.shaded.o.a.thrift</shadedPattern>
```
Comment (elahrvivaz) [inline geomesa-spark/geomesa-spark-accumulo/pom.xml]: do we care about consistent shading packages? (above we don't use abbreviations)

File: pom.xml [hunk 4]
```diff
@@ -1743,6 +1743,27 @@
...
+                <version>${project.version}</version>
+            </dependency>
+            <dependency>
+                <groupId>org.locationtech.geomesa</groupId>
+                <artifactId>geomesa-spark-core_${scala.binary.version}</artifactId>
+                <version>${project.version}</version>
+            </dependency>
+            <dependency>
+                <groupId>org.locationtech.geomesa</groupId>
+                <artifactId>geomesa-jupyter_2.11</artifactId>
```
Comment (jahhulbert-ccri) [inline pom.xml]: modify 2.11 to make it scala.binary.version

File: geomesa-spark/geomesa-spark-accumulo/pom.xml [hunk 5]
```diff
@@ -25,12 +25,22 @@
...
 
         <dependency>
             <groupId>org.locationtech.geomesa</groupId>
             <artifactId>geomesa-tools_${scala.binary.version}</artifactId>
         </dependency>
 
+        <!-- only needed for packaging into the shaded jar -->
+        <dependency>
+            <groupId>org.locationtech.geomesa</groupId>
+            <artifactId>geomesa-jupyter_2.11</artifactId>
```
Comment (jahhulbert-ccri) [inline geomesa-spark/geomesa-spark-accumulo/pom.xml]: binary version

**Comment by tkunicki** [comment]
I have updates to the this posted in https://github.com/ccri/geomesa/tree/fcr_spark_packaging

**Comment by anthonyccri** [comment]
Done

**Pull request closed without merging**